File: src/VideoBench.Domain/Entities/VideoTest.cs
using System.ComponentModel.DataAnnotations;

namespace VideoBench.Domain.Entities;

public class VideoTest
{
    [Key]
    public Guid Id { get; set; }

    public required Guid CreatedBy { get; set; }

    public required DateTime StartTime { get; set; }

    public required DateTime EndTime { get; set; }

    public int QualityId { get; set; }

    public Quality Quality { get; set; } = null!;

    public ICollection<Category> Categories { get; set; } = new List<Category>();

    public ICollection<Survey> Surveys { get; set; } = new List<Survey>();
}

File: src/VideoBench.Web/ClientApp/src/models/VideoTest.ts
import { Category } from "./Category";
import { Bitrate } from "./Bitrate";
import { UUID } from "crypto";

export interface VideoTest {
    id: UUID | null,
    startTime: string,
    endTime: string,
    qualityId: number,
    samplesNumber: number,
    bitrateValues: Bitrate[],
    categories: Category[]
}